content/en/blog/_posts/2021-01-28-PSP-deprecation.md

```diff
@@ -0,0 +1,48 @@
+
+
+## What do Pod Security Policies do now?
+
+> A Pod Security Policy is a cluster-level resource that controls security sensitive aspects of the pod | specification. The [PodSecurityPolicy](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.20/#podsecuritypolicy-v1beta1-policy) object defines a set of conditions that a pod must run with in order to be accepted into the system, as well as defaults for the related fields. They allow an administrator to control a number of security-related functions.
+
+Doc: [https://kubernetes.io/docs/concepts/policy/pod-security-policy/](https://kubernetes.io/docs/concepts/policy/pod-security-policy/)
+
+
+## Why are Pod Security Policies being deprecated? 
+
+To watch an explanation of PSP removal, you can check out this session by SIG-Auth at KubeCon NA 2019:
+
+{{< youtube "SFtHRmPuhEw?start=953" >}} 
+
+In short, PSPs are part of the kubelet and API server. The tight coupling to these parts of Kubernetes makes PSPs cumbersome both for developers and for operators. Using PSPs is often unintuitive, and setting them up too often leads to unintended results. The initial design of PSPs did not foresee many of the use cases they’ve since tried to solve.
+
+To quote the above video: "Various limitations and structural problems have prevented the PSP API from GA." Having objects in Kubernetes never reaching GA (“permabeta”) is a problem for administrators and organizations, as a non-GA feature is not considered to be ready for widespread use. Given the structural problems with the PSPs as they exist today, there is no real path for them to become GA. Per the deprecation issue, “the future of restricting pod security settings does not lie in PSP because compatibility restrictions will prevent the kinds of changes that are required.” As such, PSPs will be deprecated in Kubernetes 1.21, though they will not be fully removed until a future version.
+
+
+## What will replace them? 
+
+Securing your pods requires understanding how the pods are intended to run, and what types of behaviors or activities you seek to prevent. While Pod Security Policies have been one way to enforce security configurations, other tools in the cloud native ecosystem have security enforcement capabilities. The [Pod Security Standards](https://kubernetes.io/docs/concepts/security/pod-security-standards/) page in the Kubernetes Documentation aims “to detail recommended Pod security profiles, decoupled from any specific instantiation.”  This page is a useful reference for common pod security configurations, which can be implemented by native or external pod security enforcement tools.
```

Comment:
```suggestion
Securing your pods requires understanding how the pods are intended to run, and what types of behaviors or activities you seek to prevent. While Pod Security Policies have been one way to enforce security configurations, other tools in the cloud native ecosystem have security enforcement capabilities. The [Pod Security Standards](/pod-security-standards/) page in the Kubernetes documentation aims “to detail recommended Pod security profiles, decoupled from any specific instantiation.”  This page is a useful reference for common pod security configurations, which can be implemented by native or external pod security enforcement tools.
```